### ISSUE TITLE
`cargo clippy` fails to run if the `-- -A warnings` suffix is provided

### ISSUE THREAD

**Issue by kevinaboos** [rust-lang/rust-clippy#9012]
### Summary

When running the following on the Theseus OS code base
```sh
cargo clean && cargo clippy -Z build-std=core,alloc -Z build-std-features=compiler-builtins-mem --target cfg/x86_64-theseus.json --workspace --all-features -- -A warnings
```
We encounter the following output:
<details>
  <summary>Click to expand the output</summary>
  
  ```
    error: output of --print=file-names has changed in the compiler, cannot parse
command was: `/home/kevin/.rustup/toolchains/nightly-2022-06-17-x86_64-unknown-linux-gnu/bin/clippy-driver rustc - --crate-name ___ --print=file-names --target /tmp/Theseus/cfg/x86_64-theseus.json --crate-type bin --crate-type rlib --crate-type dylib --crate-type cdylib --crate-type staticlib --crate-type proc-macro --print=sysroot --print=cfg`

--- stdout
lib___.rlib
lib___.a
/home/kevin/.rustup/toolchains/nightly-2022-06-17-x86_64-unknown-linux-gnu
debug_assertions
feature="cargo-clippy"
panic="unwind"
proc_macro
target_abi=""
target_arch="x86_64"
target_endian="little"
target_env=""
target_feature="llvm14-builtins-abi"
target_has_atomic="16"
target_has_atomic="32"
target_has_atomic="64"
target_has_atomic="8"
target_has_atomic="ptr"
target_has_atomic_equal_alignment="16"
target_has_atomic_equal_alignment="32"
target_has_atomic_equal_alignment="64"
target_has_atomic_equal_alignment="8"
target_has_atomic_equal_alignment="ptr"
target_has_atomic_load_store="16"
target_has_atomic_load_store="32"
target_has_atomic_load_store="64"
target_has_atomic_load_store="8"
target_has_atomic_load_store="ptr"
target_os="theseus"
target_pointer_width="64"
target_vendor="unknown"
  ```
</details>

Note that when _removing_ the `-- -A warnings`, it works as expected. Funnily enough, it also works with `-- -D warnings`. Quite strange.

### Reproducer

Check out Theseus:
```
git clone --recursive --depth 1 git@github.com:theseus-os/Theseus.git
```
Then run the above command
```
cd Theseus

cargo clean && cargo clippy -Z build-std=core,alloc -Z build-std-features=compiler-builtins-mem --target cfg/x86_64-theseus.json --workspace --all-features -- -A warnings
```

Try removing the `-A warnings` or changing it to `-D warnings`, and it works as expected.

### Version

```text
As we can only build Theseus with nightly Rust, I tried several nightly versions; all 3 resulted in the same failure.
* nightly-2022-03-21
* nightly-2022-05-21
* nightly-2022-06-17

Output from the latest one: 
rustc 1.63.0-nightly (cacc75c82 2022-06-16)
binary: rustc
commit-hash: cacc75c82ebe15cf63d31034fcf7f1016cddf0e2
commit-date: 2022-06-16
host: x86_64-unknown-linux-gnu
release: 1.63.0-nightly
LLVM version: 14.0.5
```
